src/cplscheme/SerialCouplingScheme.cpp

```diff
@@ -111,14 +119,34 @@ void SerialCouplingScheme::exchangeSecondData()
     if (isImplicitCouplingScheme()) {
       receiveConvergence(getM2N());
     }
+
     PRECICE_DEBUG("Receiving data...");
+    for (const DataMap::value_type &pair : getReceiveData()) {
+      pair.second->clearTimeStepsStorage(true);
+    }
     receiveData(getM2N(), getReceiveData());
+    if (hasConverged()) {
+      // received converged result of this window, trigger move
+      for (const DataMap::value_type &pair : getReceiveData()) {
+        pair.second->moveTimeStepsStorage();
+      }
+    }
```

Comment:
Try to move this part inside receive. Possible problem: In serial coupling scheme this has to be triggered before receive for the first and after receive for the second participant.

Reply:
Still looks difficult. 0aa0a6899fc0610496c5e720388118cdf92eea50